app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -269,34 +284,84 @@ open class ProfilesViewModel(
     _tempProfilePictureBitmap.value = bitmap
   }
 
+  /**
+   * Saves the edited profile to the state.
+   *
+   * @param profile The profile to be saved.
+   */
+  fun saveEditedProfile(profile: Profile) {
+    _editedCurrentProfile.value = profile
+  }
+
   /** Clears the temporary profile picture Bitmap. */
   fun clearTempProfilePictureBitmap() {
     _tempProfilePictureBitmap.value = null
   }
 
+  /** Clears the edited profile from the state. */
+  private fun clearEditedProfile() {
+    _editedCurrentProfile.value = null
+  }
+
+  /** Resets every state related to profile edition. */
+  fun resetProfileEditionState() {
+    clearTempProfilePictureBitmap()
+    clearEditedProfile()
+    _pictureChangeState.value = ProfilePictureState.UNCHANGED
+  }
+
   /**
    * Processes the cropped image by compressing it if it is too large.
    *
    * @param bitmap The cropped Bitmap to be processed.
    */
   fun processCroppedImage(bitmap: Bitmap) {
     _tempProfilePictureBitmap.value = compressImageIfTooLarge(bitmap)
+    _pictureChangeState.value = ProfilePictureState.TO_UPLOAD
+  }
+
+  fun validateProfileChanges(context: Context) {
+    when (_pictureChangeState.value) {
+      ProfilePictureState.TO_UPLOAD ->
+          validateAndUploadProfilePicture(context) { url ->
+            val newProfile = _editedCurrentProfile.value!!.copy(profilePictureUrl = url)
+            updateProfile(newProfile)
+            _selfProfile.value = newProfile
+            // todo: "_selectedProfile.value = newProfile" needs to be removed when all the other
+            // personal profile views are updated to use the selfProfile instead of the
+            // selectedProfile
+            _selectedProfile.value = newProfile
```

Comment:
Question : why do we have to change the selected profile value ? I don't see the use here because we change the profile picture of our profile and the selected profile is used for the AroundYou page or the Notification page to see the profile of other people. Maybe I missunderstood your code

Reply:
The reason is written in the comment just above the assignment :) 
Old code in settings and `ProfileView` currently still uses `selectedProfile` to display the current users, which leads to several issues after leaving the edition screen (changes not displayed). I Fixed it in `ProfileEdit` but I didn't want to modify every other screen as it was not my task and I could have disturbed someone else's job. That's why I made this here and that I tagged it to be removed later.